pkg/metainfo/kvmetainfo/stream.go

```diff
@@ -62,11 +64,16 @@ func (stream *readonlyStream) segment(ctx context.Context, index int64) (segment
 		segment.Size = stream.info.FixedSegmentSize
 		copy(segment.EncryptedKeyNonce[:], segmentMeta.KeyNonce)
 		segment.EncryptedKey = segmentMeta.EncryptedKey
+		components := storj.SplitPath(segmentPath)
+		pointer, limits, _ = stream.db.metainfo.ReadSegment(ctx, components[1], components[2], index)
```

Comment:
This will create a bunch of orders that will never get handled. `ReadSegment` is about intending to read.

Reply:
removed changes